### PR TITLE
Add dark mode

### LINKED ISSUE #127: Ask for dark mode
need dark mode.

I use https://github.com/tolking/vuepress-theme-default-prefers-color-scheme to enable dark mode. But this plugin is not compatible with dark mode

### PR DESCRIPTION
**Pull request by mdubus**
Closes #127

## What's inside this PR ?

Dark mode support as requested in #127.

3 more changes were made : 
- Upgrade of `docs-searchbar` version inside the`package.json` (v1.3.0)
- Removal of the surrounding `.search-box` div according to [this PR](https://github.com/meilisearch/docs-searchbar.js/pull/366/files)
- Exposure of the `debug` option which was available in `docs-searchbar` but not in this plugin

### Run the playground

You can run the playground with `yarn serve` 

### Change the color palette

If you want to change the color palette, you can do so by changing the colors inside the `playground/.vuepress/styles/palette.styl` file

### Enable / disable dark mode

You can also enable/disable the dark mode (disabled by default) by adding/removing the `enableDarkMode` option in the `playground/.vuepress/config.js` file (don't forget to restart the server !): 
Capture d’écran 2021-06-23 à 08 46 27

## Screenshots

Column layout (default)
layout columns

Simple layout
layout simple

Change `$accentDarkColor` to `#a29bfe` (full change of the dark theme, cohabitation with the `vuepress-theme-default-prefers-color-scheme` plugin)
change all colors

### CODE REVIEW

**Review by curquiza — COMMENTED** [review]
Awesome!

- How is it going to be for VuePress website that does not the dark mode allowed, for example our docs?
- Does it work with https://github.com/tolking/vuepress-theme-default-prefers-color-scheme as the issue #127 mentions it?

**Comment by bidoubiwa** [comment]
```
# At the root of the plugin (vuepress-plugin-meilisearch for me):

`yarn link`

#Inside the playground folder:

rm -rf node_modules
# Remove the vuepress-plugin-meilisearch from the playground/package.json
yarn link "vuepress-plugin-meilisearch"
yarn
```

This shouln't be necessary as the playground is already connected to its root folder that contains the plugin.

**Comment by curquiza** [comment]
> There is no vuepress-plugin-meilisearch in the package.json in the playground

https://github.com/meilisearch/vuepress-plugin-meilisearch/blob/5536a3667d47a9955b2eb008a5783a7af3cc82d9/playground/package.json#L12

**Comment by bidoubiwa** [comment]
But in `config.js` it takes its root from the parent directory 

```js
module.exports = {
  title: 'Welcome to the Playground!',
  plugins: [
    [
      require('../../index.js'),
      {
        // Testing with MeiliSearch documentation content (https://docs.meilisearch.com/)
        hostUrl: 'https://docs-search-bar.meilisearch.com',
        apiKey:
          'd79226ae89f29d4dadba8d0c30c240e435f584fb83a7ae573b13eb62edec35cd',
        indexUid: 'docs'
        // "maxSuggestions": 10,
        // "placeholder": "Search as you type..."
      }
    ]
  ]
}
```
not the plugin that is in the package.json apparently but should not have any impact since it is not used in the playground

**Comment by mdubus** [comment]
> Awesome!
> 
> * How is it going to be for VuePress website that does not the dark mode allowed, for example our docs?
> * Does it work with https://github.com/tolking/vuepress-theme-default-prefers-color-scheme as the issue #127 mentions it?

- It provides the same option as in docs-searchbar `enableDarkMode`, I'm going to add this to the body of the PR :) 
- Yes it does ! I provided examples in the `screenshots` section (the screenshots in purple are the one you are looking for)

**Comment by mdubus** [comment]
> In the `config.js` from the playground: the plugin takes its root from the parent directory
> 
> ```js
> module.exports = {
>   title: 'Welcome to the Playground!',
>   plugins: [
>     [
>       require('../../index.js'),
>       {
>         // Testing with MeiliSearch documentation content (https://docs.meilisearch.com/)
>         hostUrl: 'https://docs-search-bar.meilisearch.com',
>         apiKey:
>           'd79226ae89f29d4dadba8d0c30c240e435f584fb83a7ae573b13eb62edec35cd',
>         indexUid: 'docs'
>         // "maxSuggestions": 10,
>         // "placeholder": "Search as you type..."
>       }
>     ]
>   ]
> }
> ```
> 
> not the plugin that is in the package.json. The plugin in the package.json should not have any impact in the playground since it is not used in the config file

Ok I got it. 
I didn't managed to make it work this way, so I thought they weren't linked together. The method described in the body of the PR was the only way to make it work.
I took a look at it, and the `docs-searchbar` used in the playground was the old one (without the `enableDarkMode` option), so it obviously couldn't work 😀
The problem was the `vuepress-plugin-meilisearch` in the `playground/package.json`. As it was in `v0.0.6`, it also used the wrong version of `docs-searchbar`. After removing this useless dependency, there is not problem anymore 🎉

I'm updating the body of the PR according to this. 
Thanks for pointing out this problem ! ❤️

File: playground/.vuepress/styles/palette.styl
Comment (curquiza) [inline playground/.vuepress/styles/palette.styl:8]: Can you add a new line at the end of the line? We want to keep a consistent style to avoid this kind of update in a future PR that is not related at all to this file 🙂

To avoid this kind of error, you can download the editorconfig plugin, depending on your text editor. After downloading it, every time you will save your file, it will automatically apply the basic style rules to your file (removing trailing whitespaces, adding a new line at the end of file...) thanks to the .editorconfig file in this repo.
See more about editorconfig: https://editorconfig.org/#overview
And the editorconfig file of this repo: https://github.com/meilisearch/meilisearch-ruby/blob/master/.editorconfig

**Review by curquiza — CHANGES_REQUESTED** [review]
Can you update the README according to your new feature? 🙂

File: playground/.vuepress/styles/palette.styl
Comment (mdubus) [reply playground/.vuepress/styles/palette.styl]: Done ! Indeed I didn't had the plugin in vscode. Thanks for the explanation ! ❤️

**Comment by mdubus** [comment]
> Can you update the README according to your new feature? 🙂

It's already done 🤩
I added a `Dark mode` section inside the `Customization` one

**Comment by curquiza** [comment]
My bad! I completely missed it in the middle of the files 😅

File: README.md
Comment (curquiza) [inline README.md:126]:
```suggestion
To override the default theme of the search bar, you can edit your `.vuepress/styles/palette.styl` file.<br>
```

Comment (curquiza) [reply README.md:126]: It's not convenient AT ALL, but markdown file does not consider your newline in the same paragraph. You have to put a `<br>` (or if you build markdown files, like in the documentation, download a plugin haha)

File: README.md
Comment (curquiza) [inline README.md:127]:
```suggestion
A few variables are available:
```

Comment (mdubus) [reply README.md]: Thanks for the suggestions ❤️

**Review by curquiza — APPROVED** [review]
I approve because I don't want to block the PR, but of course we need to wait for @bidoubiwa approval 🙂

@mdubus, you have write access to this repo, you don't need to work on your fork anymore 🙂

Thanks a lot!!!

**Review by bidoubiwa — CHANGES_REQUESTED** [review]
Everything looks super fine except for one thing. When you remove the following line from `playground/.vuepress/config.js`the app throws an error.
```js
  theme: 'default-prefers-color-scheme',
```

The following error is thrown: 

```
vue.runtime.esm.js?2b0e:619 [Vue warn]: Failed to resolve async component: function Layout() {
    return Promise.all(/*! import() */[__webpack_require__.e(0), __webpack_require__.e(1)]).then(__webpack_require__.bind(null, /*! ./node_modules/@vuepress/theme-default/layouts/Layout.vue */ "./node_modules/@vuepress/theme-default/layouts/Layout.vue"));
  }
Reason: Error: Module build failed (from ./node_modules/stylus-loader/index.js):
Error: vuepress-plugin-meilisearch/playground/node_modules/stylus/lib/functions/index.styl:133:3
   129| 
   130| // lighten by the given amount
   131| 
   132| lighten(color, amount)
   133|   adjust(color, 'lightness', amount)
----------^
   134| 
   135| // decrease opacity by amount
   136| 
```

**Comment by bidoubiwa** [comment]
I looked at the default palette of vuepress and it seems it is using the variable used by `default-prefers-color-scheme`. 

See [Styling in VuePress](https://vuepress.vuejs.org/config/#styling). 

Furthermore there is a [default palette](https://github.com/vuejs/vuepress/blob/master/packages/@vuepress/core/lib/client/style/config.styl) provided by vuepress. I think it is possible  to rely on these values to have default values. 

As an example, if we look at the default search bar component that is provided by vuepress it lets us [tweak with the following variable](https://vuepress.vuejs.org/plugin/official/plugin-search.html#tips).

```
$accentColor = #3eaf7c
$textColor = #2c3e50
$borderColor = #eaecef
$codeBgColor = #282c34
$arrowBgColor = #ccc
```

Because of that, I think we should be consistent with these variable naming and our variables by just adding `dark` and not use `ms`. 
It seems that we can use vuepress default theme to avoid the previous error.

**Comment by bidoubiwa** [comment]
So I tried with and without the dark theme and it worked, I'm not sure what provoked the bug the first time around: 

Update `styles/palette.style` to comply to naming convention (see previous comment)

```styl
$accentDarkColor ?= $accentColor

$inputDarkBgColor ?= #444d52

$textDarkColor ?= #eaeaea

$borderDarkColor ?= lighten($inputDarkBgColor, 10%)

$dropdownBgDarkColor ?= darken($inputDarkBgColor, 25%)
```

Play with `./vuepress/styles/palette.style`

```styl
// No need to define accentColor, works in both cases
$accentDarkColor = $accentColor // $accentColor comes from default vuepress theme see link below

$inputDarkBgColor = #444d52
$textDarkColor = #eaeaea
// $borderDarkColor needs to have inputDarkBgColor defined as information goes down to the plugin palette, so it is not aware that this variable exists since it does not exists in both the default palette and the palette from the dark-theme
$borderDarkColor = lighten($inputDarkBgColor, 10%) 
```

[default vuepress palette](https://github.com/vuejs/vuepress/blob/master/packages/@vuepress/core/lib/client/style/config.styl)

Other changes this emply:
- [ ] changing the names in MeiliSearchVueBox
- [ ] changing the names in README


Unless there is another reason not to go with these variable names?

**Comment by mdubus** [comment]
We agreed with @bidoubiwa to rename the style variable by removing the `ms-` prefix.
It's now done 🚀

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥🔥🔥

**Comment by mdubus** [comment]
bors merge

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**
